torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -1332,6 +1347,24 @@ def _test_all_reduce_helper(
                 if cuda:
                     tensor = tensor.cuda(rank_to_GPU[rank][0])
                 self.call_dist_op(":all_reduce", async_op, dist.all_reduce, tensor, op, group_id, async_op=async_op)
+                # Currently, only Gloo backend has profiling tested with CUDA enabled.
+                # Only run cuda profiling test for one rank to speed up since
+                # running with different src_rank does not affect the correctness.
+                if (
+                    src == 0
+                    and cuda
+                    and dist.get_backend() in CUDA_PROFILING_SUPPORTED_BACKENDS
+                ):
+                    self.call_dist_op(
+                        ":all_reduce",
+                        async_op,
+                        dist.all_reduce,
+                        tensor,
+                        op,
+                        group_id,
+                        async_op=async_op,
+                        profile_cuda=True,
```

Comment:
Would I be correct if I assume this will only make a difference when `cuda == True`? Is there any existing tests exploring that code path?

Reply:
`cuda` controls whether GPU tensors are used instead of CPU tensors, but `profile_cuda` controls whether profiling is enabled with CUDA or not, so they are sort of independent.